webrender/res/ps_angle_gradient.glsl

```diff
@@ -57,8 +59,18 @@ void main(void) {
 
     float offset = dot(pos - vStartPoint, vScaledDir);
 
-    oFragColor = sample_gradient(vGradientAddress,
+    vec4 color = sample_gradient(vGradientAddress,
                                  offset,
                                  vGradientRepeat);
+
+    // Un-premultiply the color from sampling the gradient.
+    color.rgb /= color.a;
```

Comment:
what happens if `color.a == 0.0`? I assume that `color.rgb` becomes INF, and then when we re-multiply by `color.a` we'll get INF * 0.0, which is not actually 0.0.

Reply:
If the answer here is "we cannot rely on this becoming 0.0 after the multiplication", then the Invert and Brightness functions will need to be fixed as well.

Reply:
AFAIK, IEEE 32bit float define INF * 0.0 = NaN, so we can't rely on division by `color.a` to be reversible

Reply:
Test in Rust - https://play.rust-lang.org/?gist=dc6143962fd7ea794266d9decd417202&version=stable

Reply:
You're right - I added an ```if``` check to only apply this if the alpha is > 0. We will also need to fix up the blend functions that @mstange mentioned - I'll do those in a separate PR.